src/test/kotlin/com/ampnet/auditornode/script/api/classes/HttpClientJavaScriptTest.kt

```diff
@@ -0,0 +1,794 @@
+                    assertEquals("header1List.get(0)", "${responseHeaders["header1"]}", header1List.get(0));
+
+                    let header2List = response.headers.get("header2");
+
+                    assertEquals("header2List.length", 1, header2List.length);
+                    assertEquals("header2List.get(0)", "${responseHeaders["header2"]}", header2List.get(0));
+
+                    assertEquals("response.cookies.length", 1, response.cookies.length);
+
+                    let cookie = response.cookies.get(0);
+
+                    assertEquals("cookie.name", "${responseCookie.name}", cookie.name);
+                    assertEquals("cookie.value", "${responseCookie.value}", cookie.value);
+                    assertEquals("cookie.domain", "${responseCookie.domain}", cookie.domain);
+                    assertEquals("cookie.path", "${responseCookie.path}", cookie.path);
+                    assertEquals("cookie.httpOnly", ${responseCookie.isHttpOnly}, cookie.httpOnly);
+                    assertEquals("cookie.secure", ${responseCookie.isSecure}, cookie.secure);
+                    assertEquals("cookie.maxAge", ${responseCookie.maxAge}, cookie.maxAge);
+                    assertEquals("cookie.sameSite", "${responseCookie.sameSite.get().name}", cookie.sameSite);
+
+                    return AuditResult.of(true);
+                }
+            """.trimIndent()
+            val result = service.evaluate(scriptSource)
+            assertThat(result).isRightContaining(AuditResult(true))
+        }
+    }
+
+    @Test
+    fun `must correctly perform post() call with some headers`() {
+        val client = mock<BlockingHttpClient>()
+        val requestUri = URI.create("http://localhost:8080")
+        val responseHeaders = mapOf(
+            "header1" to "value1",
+            "header2" to "value2"
+        )
+        val responseCookie = SimpleCookie("cookieName", "cookieValue").apply {
+            domain("example.com")
+            path("/slash")
+            httpOnly(true)
+            secure(true)
+            maxAge(1000L)
+            sameSite(SameSite.Strict)
+        }
+        val responseBody = "{}"
+        val responseCode = 200
+        val requestBody = "exampleRequest"
+
+        suppose("client will return some HTTP response") {
+            val httpRequestMatcher: (HttpRequest<String>) -> Boolean = { arg ->
+                val headersMap = arg.headers.asMap()
+                arg.uri == requestUri && arg.method == HttpMethod.POST && arg.body.get() == requestBody &&
+                    headersMap.size == 3 &&
+                    headersMap["Content-Type"]?.size == 1 && headersMap["Content-Type"]!![0] == "text/plain" &&
+                    headersMap["requestHeader1"]?.size == 1 && headersMap["requestHeader1"]!![0] == "headerValue1" &&
+                    headersMap["requestHeader2"]?.size == 1 && headersMap["requestHeader2"]!![0] == "headerValue2" &&
```

Comment:
```suggestion
                    headersMap["Content-Type"]?.size == 1 && headersMap["Content-Type"]?.get(0) == "text/plain" &&
                    headersMap["requestHeader1"]?.size == 1 && headersMap["requestHeader1"]?.get(0) == "headerValue1" &&
                    headersMap["requestHeader2"]?.size == 1 && headersMap["requestHeader2"]?.get(0) == "headerValue2" &&
```

Try to avoid using `!!`, there should always be a solution with `?` :)